psydac/linalg/stencil.py

```diff
@@ -763,7 +767,7 @@ def __add__(self, m):
         assert m._domain   is self._domain
         assert m._codomain is self._codomain
         assert m._pads     == self._pads
-        w = StencilMatrix(self._domain, self._codomain, self._pads)
+        w = StencilMatrix(self._domain, self._codomain, self._pads, self._backend)
```

Comment:
Can you please check if the two matrices use the same backend, and if they don't raise a clear warning like the following?

```python
import warnings

if m._backend != self._backend:
    msg = 'Adding two matrices with different backends is ambiguous - defaulting to backend of first addend'
    warnings.warn(msg, category=RuntimeWarning)
```